Makefile

```diff
@@ -62,58 +101,77 @@ GRUB_MODULES = \
 	raid6rec \
 	video
 
-# Download the latest version of package $1 for architecture $(ARCH), unpacking
+all: boot install
+
+# Check if we're running under snapcraft. If not, we need to 'stage' some
+# packages by ourselves.
+ifdef SNAPCRAFT_PROJECT_NAME
+stage-package:
+	$(info Skipping staging of package $(package) under snapcraft)
+else
+# Download the latest version of package $package for architecture $(ARCH), unpacking
 # it into $(STAGEDIR). For example, the following invocation will download the
 # latest version of u-boot-rpi for armhf, and unpack it under STAGEDIR:
 #
-#  $(call stage_package,u-boot-rpi)
+#  $(MAKE) stage-package package=u-boot-rpi
 #
-define stage_package
+stage-package:
+	$(info Staging package $(package)...)
 	mkdir -p $(STAGEDIR)/tmp
-	( \
-		cd $(STAGEDIR)/tmp && \
-		apt-get download \
-			-o APT::Architecture=$(ARCH) $$( \
-				apt-cache \
-					-o APT::Architecture=$(ARCH) \
-					showpkg $(1) | \
-					sed -n -e 's/^Package: *//p' | \
-					sort -V | tail -1 \
-			); \
-	)
-	dpkg-deb --extract $$(ls $(STAGEDIR)/tmp/$(1)*.deb | tail -1) $(STAGEDIR)
-endef
-
-all: boot install
-
-boot:
-	# Check if we're running under snapcraft. If not, we need to 'stage'
-	# some packages by ourselves.
-ifndef SNAPCRAFT_PROJECT_NAME
-	$(call stage_package,grub-pc-bin)
-	$(call stage_package,grub-efi-amd64-signed)
-	$(call stage_package,shim-signed)
+	# setup chdist APT environment for SERIES-ARCH and run apt update
+	if [ ! -d  $(STAGEDIR)/tmp/chdist ]; then \
```

Comment:
You only really need this if SERIES or ARCH does not match the environment you're running in? Or is there some reason to do this then too?

Reply:
Correct; which is currently what happens if you're building a lunar/mantic image from a jammy host using ubuntu-images definitions: ubuntu-image pulls the gadget repo, runs make SERIES=lunar ARCH=arm64, but SERIES was being completely ignored and you ended up with jammy's grub binaries in your lunar image.